operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependent.java

```diff
@@ -10,6 +10,10 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.TYPE})
 public @interface KubernetesDependent {
+  String SAME_AS_PARENT = "JOSDK_SAME_AS_PARENT";
+  String WATCH_ALL_NAMESPACES = "JOSDK_ALL_NAMESPACES";
```

Comment:
This should not be in the `@ControllerConfiguration` ? 
That might be unrelated or not strictly related, but probably we should have just to watch all or selected namespaces. Not sure about the current namespace, but that should be also explicit in case.

Reply:
No, this needs to be on this annotation for the dependent to be able to force some configurations which would otherwise be impossible to express.